src/components/AccountSeed.js

```diff
@@ -17,154 +17,93 @@
+    // the word list is sorted, get the index we should start searching
+    // for the word
+    let fromIndex = word_list.findIndex(w => w.startsWith(input));
     if (fromIndex === -1) {
       return [];
     }
-    if (WORDS[fromIndex] === input) {
-      fromIndex = 0;
-    }
+
     const SUGGESTIONS_COUNT = 5;
     const result = [];
     let yielded = 0;
-    result.push(WORDS[fromIndex]);
-    while (yielded < SUGGESTIONS_COUNT && WORDS[fromIndex] !== undefined) {
-      ++fromIndex;
-      if (!WORDS[fromIndex].startsWith(input)) {
+    while (yielded < SUGGESTIONS_COUNT && fromIndex < word_list.length) {
+      if (!word_list[fromIndex].startsWith(input)) {
         return result;
       }
-      if (words.indexOf(WORDS[fromIndex]) !== -1) {
-        continue;
+
+      // do not suggest words that where already used (and part of `inputWordArray`)
+      // unless it's the latest word e.g what the user is typing
+      if (inputWordArray.indexOf(word_list[fromIndex]) === -1 || inputWordArray.indexOf(word_list[fromIndex]) === (inputWordArray.length - 1)) {
+        result.push(word_list[fromIndex]);
+        yielded++
       }
-      result.push(WORDS[fromIndex]);
-      ++yielded;
+      fromIndex++;
     }
     return result;
   }
 
-  renderSuggestions() {
-    const { value } = this.props;
-    const words = value.length ? value.split(' ') : [];
-    const wordPosition = this.getWordPosition();
-    let searchInput = this.getSearchInput();
-    if (WORDS_INDEX[searchInput]) {
-      searchInput = '';
+  narrowWordList = (inputWordArray) => {
+    // if the last word is only present in BIP39 word list
+    if (BIP39_WORDS.indexOf(inputWordArray[inputWordArray.length - 2]) !== -1 && PARITY_WORDS.indexOf(inputWordArray[inputWordArray.length - 2]) === -1) {
+      this.setState({ useParityWordList: false })
+      // if the second last word is only present in Parity word list
+    } else if (BIP39_WORDS.indexOf(inputWordArray[inputWordArray.length - 2]) === -1 && PARITY_WORDS.indexOf(inputWordArray[inputWordArray.length - 2]) !== -1) {
+      this.setState({ useBIP39WordList: false })
     }
-    const suggestions = this.getSuggestions(searchInput, words);
+  }
+
+  generateSuggestions = (inputWordArray) => {
+    const { useBIP39WordList, useParityWordList } = this.state;
+
+    // try to narrow down the word list using the last word typed, 
+    // as soon as a second word is being typed
+    inputWordArray.length > 1 && useBIP39WordList && useParityWordList && this.narrowWordList(inputWordArray);
+
+    let suggestions = []
+    useParityWordList && suggestions.push(...this.getSuggestions(inputWordArray, PARITY_WORDS));
+    useBIP39WordList && suggestions.push(...this.getSuggestions(inputWordArray, BIP39_WORDS));
+
+    //return a deduplicated sorted array if both word lists are still used
+    return (useBIP39WordList && useParityWordList) ? [...new Set(suggestions.sort())] : suggestions;
```

Comment:
```js
// de-duplicate results if both lists were used
if (useBIP39WordList && useParityWordList) {
  suggestions = Array.from(new Set(suggestions)).sort();
}

return suggestions;
```

More verbose, but more readable. Also, no need to sort the results before creating the set, I don't think the Set necessarily has any guarantees about insertion order (although it likely does have that behavior in V8).